### PR TITLE
CRI: add docs for AppArmor and Seccomp

### PR DESCRIPTION
**Pull request by feiskyer**
This PR adds docs for AppArmor and Seccomp in CRI.

cc/ @yujuhong @Random-Liu

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34819)

### CODE REVIEW

**Comment by yujuhong** [comment]
@timstclair could you help review this? Thanks!

File: pkg/kubelet/api/v1alpha1/runtime/api.pb.go [hunk 1]
```diff
@@ -566,7 +566,35 @@ type PodSandboxConfig struct {
...
+	//        http://wiki.apparmor.net/index.php/AppArmor_Core_Policy_Reference
+	//
+	// 2. Seccomp
+	//
+	//      key: security.alpha.kubernetes.io/seccomp/pod
+	//      description: the seccomp profile for the containers of an entire pod.
+	//      value: see below.
+	//
+	//      key: security.alpha.kubernetes.io/seccomp/container/<container name>
+	//      description: the seccomp profile for the container.
```
Comment (timstclair) [inline pkg/kubelet/api/v1alpha1/runtime/api.pb.go]: Add `(overides pod)`?

File: pkg/kubelet/api/v1alpha1/runtime/api.proto [hunk 2]
```diff
@@ -187,7 +187,35 @@ message PodSandboxConfig {
     // Labels are key value pairs that may be used to scope and select individual resources.
     map<string, string> labels = 6;
     // Annotations is an unstructured key value map that may be set by external
-    // tools to store and retrieve arbitrary metadata.
+    // tools to store and retrieve arbitrary metadata. Runtimes should support
```
Comment (timstclair) [inline pkg/kubelet/api/v1alpha1/runtime/api.proto]: > Runtimes should support...

I think both of these features should be considered optional (nice to haves).

**Review by timstclair — COMMENTED** [review]
Docs LGTM, but do we really want to get into the business of enumerating every annotation that the runtime could care about? It sounds like features like these won't be annotations in the near future (discussion [here](https://github.com/kubernetes/kubernetes/issues/30819)), so this might not be a big deal.

File: pkg/kubelet/api/v1alpha1/runtime/api.pb.go (hunk 1 above)
Comment (feiskyer) [inline pkg/kubelet/api/v1alpha1/runtime/api.pb.go]: ack.

File: pkg/kubelet/api/v1alpha1/runtime/api.proto (hunk 2 above)
Comment (feiskyer) [inline pkg/kubelet/api/v1alpha1/runtime/api.proto]: ack

**Comment by feiskyer** [comment]
> Docs LGTM, but do we really want to get into the business of enumerating every annotation that the runtime could care about? It sounds like features like these won't be annotations in the near future (discussion here), so this might not be a big deal.

It's better to document these annotations in CRI clearly, so runtime maintainers don't need to lookup those annotations from kubernetes API. We can move those features to sandbox/container configs after #30819 .

@timstclair Addressed comments. PTAL.

**Comment by feiskyer** [comment]
@k8s-bot cvm gke e2e test this

**Comment by feiskyer** [comment]
@timstclair Comments addressed. PTAL.

**Comment by timstclair** [comment]
LGTM

**Comment by feiskyer** [comment]
@k8s-bot gci gke e2e test this

**Pull request merged**